arches/app/media/js/utils/create-vue-application.js

```diff
@@ -0,0 +1,38 @@
+import PrimeVue from 'primevue/config';
+
+import { createApp } from 'vue';
+import { createGettext } from "vue3-gettext";
+
+export default async function createVueApp(originEntityName, vueComponent){
+    /**
+     * This wrapper allows us to maintain a level of control inside arches-core
+     * over Vue apps. For instance this allows us to abstract i18n setup/config
+     * away from app initialization, and also allows us to ensure any plugins 
+     * we'd like to use across the Arches ecosystem will be available. This also
+     * Vue apps more easily extensible if we choose to add plugins or logic in
+     * the future.
+     **/ 
+
+    /**
+     * TODO: cbyrd #10501 - we should add an event listener that will re-fetch i18n data
+     * and rebuild the app when a specific event is fired from the LanguageSwitcher component.
+    **/ 
+    return fetch(`/api/get_frontend_i18n_data?originEntityName=${originEntityName}`).then(resp => {
```

Comment:
we should remove the leading slash from the fetch call so that the url isn't rooted at the base host.  This then allows for urls that include the language code in the url itself (which is one of the ways in which the get_language() method picks up the language code).

Reply:
fixed in latest